### PR TITLE
A new place for themes 🥳 

### PR DESCRIPTION
**Pull request by inderpreetsingh**
screenshot

### CODE REVIEW

File: www/index.html [hunk 1]
```diff
@@ -66,7 +66,7 @@
       </ul>
     </div>
     <div id="app-frame">
-      <webview id="main-viewer" src="viewer.html" nodeintegration class="base-ui"></webview>
+      <webview id="main-viewer" nodeintegration="true" src="viewer.html" class="base-ui"> </webview>
```
Comment (tsingh777) [inline www/index.html:69]: is this needed? the presence alone should make the attr value true.

File: www/main/settings/components/ThemeContainer.jsx [hunk 2]
```diff
@@ -0,0 +1,73 @@
...
+                theme={theme}
+              >
+                {i18n.t(`THEMES.${theme.name}`)}
+              </Tile>
+            ))}
+          </React.Fragment>
+        ))}
+        <header className="options-header">{i18n.t(`THEMES.CUSTOM_BACKGROUNDS`)}</header>
+        <label className="file-input-label">
+          New Image
```
Comment (tsingh777) [inline www/main/settings/components/ThemeContainer.jsx:62]: should come from i18n.

File: www/main/theme_editor.js [hunk 3]
```diff
@@ -1,3 +1,4 @@
+/* eslint-disable no-unused-vars */
```
Comment (tsingh777) [inline www/main/theme_editor.js:1]: ?

File: www/main/theme_editor.js [hunk 4]
```diff
@@ -284,13 +285,14 @@ const imageInput = themesContainer =>
...
 module.exports = {
+  themes,
   defaultTheme,
   updateCeremonyThemeTiles,
   getCurrentTheme,
   getTheme,
   applyTheme,
   init() {
-    const themeOptions = document.querySelector('#custom-theme-options');
+    /* const themeOptions = document.querySelector('#custom-theme-options');
```
Comment (tsingh777) [inline www/main/theme_editor.js:295]: remove?

File: www/main/settings/components/SettingViewer.jsx [hunk 5]
```diff
@@ -0,0 +1,157 @@
...
+  const nextLineStyles = {
+    fontSize: `${gurbaniFontSize * 3}px`,
+    opacity: 0.5,
+    display: displayNextLine ? 'block' : 'none',
+  };
+
+  return (
+    <div className={`settings-viewer theme-${theme}`} style={commonStyles}>
+      <h1 className="gurbani gurmukhi" style={gurbaniStyles}>
+        <div>
```
Comment (tsingh777) [inline www/main/settings/components/SettingViewer.jsx:44]: image
That doesn't look right.

**Review by tsingh777 — COMMENTED** [review]
Theme seems to be only applying to viewer

**Comment by inderpreetsingh** [comment]
Looks like I messed the branch up during merge conflicts :| , i will fix these things asap.

File: www/main/settings/components/SettingViewer.jsx [hunk 6]
```diff
@@ -0,0 +1,157 @@
+import React from 'react';
+import { useStoreState } from 'easy-peasy';
+
+const SettingViewer = () => {
```
Comment (tsingh777) [inline www/main/settings/components/SettingViewer.jsx:4]: Think the new window is causing some compactness. 
image

File: www/main/settings/components/ThemeContainer.jsx [hunk 7]
```diff
@@ -0,0 +1,73 @@
...
+              >
+                {i18n.t(`THEMES.${theme.name}`)}
+              </Tile>
+            ))}
+          </React.Fragment>
+        ))}
+        <header className="options-header">{i18n.t(`THEMES.CUSTOM_BACKGROUNDS`)}</header>
+        <label className="file-input-label">
+          {i18n.t(`THEMES.NEW_IMAGE`)}
+          <input className="file-input" id="themebg-upload" type="file" />
```
Comment (tsingh777) [inline www/main/settings/components/ThemeContainer.jsx:63]: not sure if this is applying. does it need an on change handler. @maneetpaul is this a used feature that we should continue supporting?

File: www/main/navigator/Navigator.jsx [hunk 8]
```diff
@@ -38,9 +38,6 @@ const Navigator = () => {
               >
                 <i className="fa fa-clock-o"></i> <span data-key="HISTORY"></span>
               </span>
-              <span className="nav-header-tab nav-header-text" id="themes-tab" data-title="themes">
-                <i className="fa fa-paint-brush"></i> <span data-key="THEMES"></span>
```
Comment (tsingh777) [inline www/main/navigator/Navigator.jsx:42]: image
Option still available under single display.

File: www/main/settings/components/Settings.jsx [hunk 9]
```diff
@@ -5,15 +5,22 @@ import { Overlay } from '../../common/sttm-ui';
 
 import SettingsNav from './SettingsNav';
 import SettingsContainer from './SettingsContainer';
-
+import ThemeContainer from './ThemeContainer';
+import SettingViewer from './SettingViewer';
 import { settingsNavObj, settingsObj } from '../utils';
 
 const Settings = ({ onScreenClose }) => {
```
Comment (tsingh777) [inline www/main/settings/components/Settings.jsx:12]: Think we need to improve the CSS for smaller view ports 
image

File: www/main/settings/components/ThemeContainer.jsx [hunk 10]
```diff
@@ -0,0 +1,73 @@
...
+  { type: 'COLOR', title: 'COLORS' },
+  { type: 'BACKGROUND', title: 'BACKGROUNDS' },
+  { type: 'SPECIAL', title: 'SPECIAL_CONDITIONS' },
+];
+
+const ThemeContainer = () => {
+  const { theme: currentTheme } = useStoreState(state => state.userSettings);
+  const { setTheme } = useStoreActions(state => state.userSettings);
+
+  const applyTheme = (themeInstance, isCustom) => {
```
Comment (tsingh777) [inline www/main/settings/components/ThemeContainer.jsx:23]: Might need to pull the hot fix we did into the branch, don't think settings are applying properly to extended display.

File: www/main/settings/components/demo.html [hunk 11]
```diff
@@ -0,0 +1,46 @@
+<div id="slide10" class="slide">
+  <h1 class="gurbani gurmukhi">
```
Comment (tsingh777) [inline www/main/settings/components/demo.html:2]: not sure the fonts are applying properly in the preview window. The Gurbani doesn't have the same font weight and such as the main presenter window.

File: www/main/settings/components/SettingViewer.jsx (hunk 6 above)
Comment (inderpreetsingh) [reply www/main/settings/components/SettingViewer.jsx:4]: @tsingh777 @maneetpaul we need to figure out what's our min supported resolution. I am facing similar problems with new designs for search pane as well.

File: www/main/settings/components/demo.html [hunk 12]
```diff
@@ -0,0 +1,46 @@
+<div id="slide10" class="slide">
```
Comment (tsingh777) [inline www/main/settings/components/demo.html:1]: image
The background image isn't centered

File: www/main/settings/components/ThemeContainer.jsx (hunk 7 above)
Comment (maneetpaul) [reply www/main/settings/components/ThemeContainer.jsx:63]: Yes, we should maintain this feature

File: www/main/settings/components/SettingViewer.jsx [hunk 13]
```diff
@@ -0,0 +1,157 @@
...
+
+  return (
+    <div className={`settings-viewer theme-${theme}`} style={commonStyles}>
+      <h1 className="gurbani gurmukhi" style={gurbaniStyles}>
+        <div>
+          <span className="padchhed">
+            <span>ik®pw</span>
+            <span>kry</span>
+            <span>jy</span>
+            <span className="visraam-main visraam-sttm2">AwpxI</span>
```
[the remark this answers is not in the document]
Comment (tsingh777) [reply www/main/settings/components/SettingViewer.jsx:52]: This still needs to be resolved.

File: www/main/settings/components/ThemeContainer.jsx [hunk 14]
```diff
@@ -0,0 +1,73 @@
...
+      <div id="custom-theme-options">
+        {themeTypes.map(({ type, title }) => (
+          <React.Fragment key={type}>
+            <header className="options-header">{i18n.t(`THEMES.${title}`)}</header>
+            {groupThemes(type).map(theme => (
+              <Tile
+                key={theme.name}
+                onClick={() => {
+                  if (currentTheme !== theme.key) {
+                    applyTheme(theme);
```
[the remark this answers is not in the document]
Comment (tsingh777) [reply www/main/settings/components/ThemeContainer.jsx:68]: This still needs to be resolved.

**Review by tsingh777 — APPROVED** [review]
2 issues still. If its easier to merge this and address them as follow up, I approve.

File: www/main/settings/utils/apply-theme.js [hunk 15]
```diff
@@ -0,0 +1,30 @@
+import { remote } from 'electron';
+
+const { store } = remote.require('./app');
+const analytics = remote.getGlobal('analytics');
+
+export const applyTheme = (themeInstance, isCustom, setTheme, setThemeBg) => {
+  if (!isCustom) {
+    setTheme(themeInstance.key);
```
Comment (tsingh777) [inline www/main/settings/utils/apply-theme.js:8]: image
Still not applying to ceremonies "current theme" option

File: www/main/settings/components/Settings.jsx [hunk 16]
```diff
@@ -5,15 +5,22 @@ import { Overlay } from '../../common/sttm-ui';
...
   return (
     <Overlay onScreenClose={onScreenClose}>
```
Comment (tsingh777) [inline www/main/settings/components/Settings.jsx:14]: I don't like how the settings aren't fluid. They're a static size that get scroll bars. We can do better than that on smaller screen.

**Review by tsingh777 — DISMISSED** [review]
Some concerns we can follow up on.

File: www/main/settings/utils/apply-theme.js (hunk 15 above)
Comment (inderpreetsingh) [reply www/main/settings/utils/apply-theme.js:8]: Oops sorry this is fixed, pushing commmit now.. :(

File: www/main/settings/components/Settings.jsx (hunk 16 above)
Comment (inderpreetsingh) [reply www/main/settings/components/Settings.jsx:14]: @maneetpaul @prebhdevsingh  
possible to get designs for this.. I can send some options as well.. 🙏

File: www/main/common/sttm-ui/custom-tile/CustomBgTile.jsx [hunk 17]
```diff
@@ -0,0 +1,32 @@
+import React from 'react';
+import PropTypes from 'prop-types';
+
+const CustomBgTile = ({ customBg, onApply, onRemove }) => {
```
Comment (tsingh777) [inline www/main/common/sttm-ui/custom-tile/CustomBgTile.jsx:4]: image
- In production we show the thumbnail of the image but this change shows blank image. 
- Also in prod we can't delete the image unless we're on a different theme. We can delete the custom theme anytime here. 
- The custom theme isn't showing up in the ceremonies selection for current theme.

**Review by tsingh777 — APPROVED** [review]

**Review by tsingh777 — APPROVED** [review]

**Pull request merged**